api/src/main/java/org/pmiops/workbench/workspaceadmin/WorkspaceAdminServiceImpl.java

```diff
@@ -403,18 +405,120 @@ public DbWorkspace setPublished(
       String workspaceNamespace, String firecloudName, boolean publish) {
     final DbWorkspace dbWorkspace = workspaceDao.getRequired(workspaceNamespace, firecloudName);
 
+    updateWorkspaceACL(
+        dbWorkspace.getWorkspaceNamespace(), dbWorkspace.getFirecloudName(), publish);
+
+    dbWorkspace.setPublished(publish);
+    return workspaceDao.saveWithLastModified(dbWorkspace, userProvider.get());
+  }
+
+  @Override
+  public void publishWorkspaceViaDB(
+      String workspaceNamespace, PublishWorkspaceRequest publishWorkspaceRequest) {
+    final DbWorkspace dbWorkspace = workspaceDao.getByNamespace(workspaceNamespace).orElseThrow();
+
+    Optional<DbFeaturedWorkspace> featuredWorkspace =
+        featuredWorkspaceDao.findByWorkspace(dbWorkspace);
+
+    DbFeaturedWorkspace dbFeaturedWorkspace;
+    if (featuredWorkspace.isPresent()) {
+      // If featuredWorkspace is Present check if category is same as requested or not
+      // If yes: do nothing and return
+      // If no: update featuredWorkspace with new category
+      dbFeaturedWorkspace = featuredWorkspace.get();
+      if (dbFeaturedWorkspace
```

Comment:
this would also be better handled with the mapper.  then you can check for direct equality.